subprojects/docs/src/docs/userguide/publishingOverview.adoc

```diff
@@ -0,0 +1,263 @@
+
+NOTE: You may define one, and only one, repository without a name. This translates to an implicit name of "Maven" for Maven repositories and "Ivy" for Ivy repositories. All other repository definitions must be given an explicit name.
+
+In combination with the project's `group` and `version`, the publication and repository definitions provide everything that Gradle needs to publish the project's production JAR. Gradle will then create a dedicated `publishMyLibraryPublicationToMyRepoRepository` task that does just that. Its name is based on the template `publish__PubName__PublicationTo__RepoName__Repository`. See the appropriate publishing plugin's documentation for more details on the nature of this task and any other tasks that may be available to you.
+
+You can either execute the individual publishing tasks directly, or you can execute `publish`, which will run all the available publishing tasks. In this example, `publish` will just run `publishMyLibraryPublicationToMavenRepository`.
+
+[NOTE]
+====
+Basic publishing to an Ivy repository is very similar: you simply use the Ivy Publish Plugin, replace  `MavenPublication` with api:org.gradle.api.publish.ivy.IvyPublication[], and use `ivy` instead of `maven` in the repository definition.
+
+There are differences between the two types of repository, particularly around the extra metadata that each support — for example, Maven repositories require a POM file while Ivy ones have their own metadata format — so see the plugin chapters for comprehensive information on how to configure both publications and repositories for whichever repository type you're working with.
+====
+
+That's everything for the basic use case. However, many projects need more control over what gets published, so we look at several common scenarios in the following sections.
+
+[[sec:publishing_custom_artifacts_to_maven]]
+=== Adding custom artifacts to a publication
+
+Users often need to include additional artifacts with a publication, one of the most common examples being that of "-sources" and "-javadoc" JARs for JVM libraries. This is easy to do for both Maven- and Ivy-compatible repositories via the `artifact` configuration.
+
+The following sample configures "-sources" and "-javadoc" JARs for a Java project and attaches them to the main (Maven) publication, i.e. the production JAR:
+
+++++
+<sample dir="maven-publish/javaProject" id="publishing_maven:publish-custom-artifact" title="Adding an additional archive artifact to a MavenPublication">
+    <sourcefile file="build.gradle" snippet="publish-custom-artifact"/>
+</sample>
+++++
+
+There are several important things to note about the sample:
+
+ * The `artifact()` configuration accepts archive tasks as an argument — like `sourcesJar` in the sample — as well as any type of argument accepted by api:org.gradle.api.Project#file(java.lang.Object)[], such as a `File` instance or string file path.
```

Comment:
The `artifact()` method. The word "configuration" is already heavily overloaded :)

Reply:
Indeed it is. I would still use "configuration", "settings" or "options" when talking about properties and methods within an `artifact {}` block. I'm becoming quite frustrated with the use of "configuration" for a dependency/artifact collection. I guess it comes from the Ivy terminology, but it's not a great name sadly.